From f03edc78f4f3695af4d6dff20969e5d573d4a293 Mon Sep 17 00:00:00 2001
From: Simon Xiao <sixiao@microsoft.com>
Date: Fri, 14 Apr 2017 14:42:58 -0700
Subject: hv_netvsc: change netvsc device default duplex to FULL
Reply-To: joseph.salisbury@microsoft.com

The netvsc device supports full duplex by default.
This warnings in log from bonding device which did not like
seeing UNKNOWN duplex.

Signed-off-by: Simon Xiao <sixiao@microsoft.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
(cherry picked from commit f3c9d40ee12926f330a1dfebce0bebadd1406ba6)
Signed-off-by: Joseph Salisbury <joseph.salisbury@microsoft.com>
---
 drivers/net/hyperv/netvsc_drv.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/net/hyperv/netvsc_drv.c b/drivers/net/hyperv/netvsc_drv.c
index a8f8f57375bb..ce7a4a442df4 100644
--- a/drivers/net/hyperv/netvsc_drv.c
+++ b/drivers/net/hyperv/netvsc_drv.c
@@ -818,7 +818,7 @@ static void netvsc_init_settings(struct net_device *dev)
 	struct net_device_context *ndc = netdev_priv(dev);
 
 	ndc->speed = SPEED_UNKNOWN;
-	ndc->duplex = DUPLEX_UNKNOWN;
+	ndc->duplex = DUPLEX_FULL;
 }
 
 static int netvsc_get_settings(struct net_device *dev, struct ethtool_cmd *cmd)
-- 
2.17.1

